### ISSUE TITLE
capped_distance functions return distances larger then max_dist

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/cellgrid#12]
See https://gist.github.com/kain88-de/247049819f75c0a72b3d61c7967f10ec

The max_dist is set to 10 A.U but there are distances returned with a larger value. The docs state that this shouldn't happen.

**Comment by kain88-de** [comment]
The reason why this happens is because `max_dist` used to build the cellgrid binsize. For two particles in neighboring cells it is possible to have a distance larger then the cell bin size.

**Comment by richardjgowers** [comment]
Yeah so the max_dist is slightly approximate right now which is wrong.  An easy fix would be `dists = dists[dists <= max_dist]` right at the end.  Or alternatively filter them as you insert.
